prac_06/guitars.py

```diff
@@ -1,19 +1,75 @@
+"""a program that keeps a list of user guitars and then displayes them once the user enters an empty string"""
 
-VINTAGE_AGE = 50
+from guitar import Guitar
 
 
-class Guitars:
+def main():
+    """get guitars and add them to a list until nothing is entered then display all guitars """
+    guitars = []
+    print("My Guitars!")
+    name, year, cost = get_input()
+    guitar = Guitar(name, cost, year)
+    while guitar.name != "":
+        print("{} added!".format(guitar))
+        guitars.append(guitar)
+        name, year, cost = get_input()
+        guitar = Guitar(name, cost, year)
+    print("\n_______Snip________\n")
+    display_guitars(guitars)
 
-    def __init__(self, name="", cost=0, year=0):
-        self.cost = cost
-        self.name = name
-        self.year = year
 
-    def __str__(self):
-        return "{} ({}) : {:.2}".format(self.name, self.year, self.cost)
+def display_guitars(guitars):
+    """display all guitars"""
+    for i, guitar in enumerate(guitars, 1):
+        vintage_string = ""
+        if guitar.is_vintage():
+            vintage_string = "(vintage)"
+        print("Guitar {}: {:>20} ({}), worth ${:10,.2f}{}".format(i, guitar.name, guitar.year, guitar.cost,
+                                                                  vintage_string))
+    else:
+        print("no Guitars")
 
-    def get_age(self):
-        return 2021 - self.year
 
-    def is_vintage(self):
-        return self.get_age() >= VINTAGE_AGE
+def get_input():
+    """get all inputs for guitar class"""
+    name = input("name: ")
+    if name == "":
+        return "", 0, 0
+    year = get_valid_int("year: ")
+    cost = get_valid_float("cost: ")
+    return name, year, cost
+
+
+def get_valid_float(prompt):
+    """Get a valid float."""
+    is_valid_input = False
+    number = 0
+    while not is_valid_input:
+        try:
+            number = float(input(prompt))
+            if number <= 0:
+                print("Number must be > 0")
+            else:
+                is_valid_input = True
+        except ValueError:
+            print("invalid input; Enter a valid number")
+    return number
+
```

Comment:
get_valid_float() and get_valid_int() are very similar functions
it might be worth trying to create a single function for this purpose that takes a parameter
of whether to require a float or an int